File: Práctica 3 (pilas y colas)/Pila.java
class Pila {
    private long[] arreglo;
    private int top;
    private int n;

    public Pila(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.top = -1;
    }

    public void push(long e) {
        if (!isFull()) {
            arreglo[++top] = e;
        }
    }

    public Long pop() {
        if (!isEmpty()) {
            return arreglo[top--];
        }
        return null; // Si está vacía
    }

    public Long peek() {
        if (!isEmpty()) {
            return arreglo[top];
        }
        return null; // Si está vacía
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }
}

public class Main {
    public static void main(String[] args) {
        Pila pila = new Pila(5);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        pila.push(40);
        System.out.println("Tope de la pila: " + pila.peek()); 
        System.out.println("Elemento desapilado: " + pila.pop()); 
        System.out.println(pila.isEmpty());
        System.out.println(pila.isFull());
    }
}

File: Práctica 3 (pilas y colas)/Pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [0] * n  
        self.top = -1
        self.n = n

    def push(self, e):
        if not self.isFull():
            self.top += 1
            self.arreglo[self.top] = e

    def long_pop(self):
        if not self.isEmpty():
            valor = self.arreglo[self.top]
            self.top -= 1
            return valor
        return None 

    def long_peek(self):
        if not self.isEmpty():
            return self.arreglo[self.top]
        return None  

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

pila = Pila(5)
pila.push(20)
pila.push(10)
pila.push(30)
print("Tope de la pila:", pila.long_peek()) 
print("Elemento desapilado:", pila.long_pop())
